internal/api/ui/login/static/resources/scripts/password_policy_check.js

```diff
@@ -1,77 +1,71 @@
-function ComplexityPolicyCheck(policyElement, pwNew, pwNewConfirmation) {
-    let minLength = policyElement.dataset.minlength;
-    let upperRegex = policyElement.dataset.hasUppercase;
-    let lowerRegex = policyElement.dataset.hasLowercase;
-    let numberRegex = policyElement.dataset.hasNumber;
-    let symbolRegex = policyElement.dataset.hasSymbol;
+function ComplexityPolicyCheck(passwordElement) {
+  let minLength = passwordElement.dataset.minlength;
+  let upperRegex = passwordElement.dataset.hasUppercase;
+  let lowerRegex = passwordElement.dataset.hasLowercase;
+  let numberRegex = passwordElement.dataset.hasNumber;
+  let symbolRegex = passwordElement.dataset.hasSymbol;
 
-    let invalid = 0;
+  let invalid = 0;
 
-    let minlengthelem = document.getElementById('minlength');
-    if (pwNew.length >= minLength) {
-        ValidPolicy(minlengthelem);
+  let minlengthelem = document.getElementById("minlength");
+  if (passwordElement.value.length >= minLength) {
+    ValidPolicy(minlengthelem);
+  } else {
+    InvalidPolicy(minlengthelem);
+    invalid++;
+  }
+  let upper = document.getElementById("uppercase");
+  if (upperRegex !== "") {
+    if (RegExp(upperRegex).test(passwordElement.value)) {
+      ValidPolicy(upper);
     } else {
-        InvalidPolicy(minlengthelem);
-        invalid++;
+      InvalidPolicy(upper);
+      invalid++;
     }
-    let upper = document.getElementById('uppercase');
-    if (upperRegex !== "") {
-        if (RegExp(upperRegex).test(pwNew)) {
-            ValidPolicy(upper);
-        } else {
-            InvalidPolicy(upper);
-            invalid++;
-        }
-    }
-    let lower = document.getElementById('lowercase');
-    if (lowerRegex !== "") {
-        if (RegExp(lowerRegex).test(pwNew)) {
-            ValidPolicy(lower);
-        } else {
-            InvalidPolicy(lower);
-            invalid++;
-        }
-    }
-    let number = document.getElementById('number');
-    if (numberRegex !== "") {
-        if (RegExp(numberRegex).test(pwNew)) {
-            ValidPolicy(number);
-        } else {
-            InvalidPolicy(number);
-            invalid++;
-        }
+  }
+  let lower = document.getElementById("lowercase");
+  if (lowerRegex !== "") {
+    if (RegExp(lowerRegex).test(passwordElement.value)) {
+      ValidPolicy(lower);
+    } else {
+      InvalidPolicy(lower);
+      invalid++;
     }
-    let symbol = document.getElementById('symbol');
-    if (symbolRegex !== "") {
-        if (RegExp(symbolRegex).test(pwNew)) {
-            ValidPolicy(symbol);
-        } else {
-            InvalidPolicy(symbol);
-            invalid++;
-        }
+  }
+  let number = document.getElementById("number");
+  if (numberRegex !== "") {
+    if (RegExp(numberRegex).test(passwordElement.value)) {
+      ValidPolicy(number);
+    } else {
+      InvalidPolicy(number);
+      invalid++;
     }
```

Comment:
IMO the code to check the policy is the same for all policies. If so please reuse the same logic instead of copy pasta

Reply:
same above